packages/shared/src/hooks/useUserMention.ts

```diff
@@ -0,0 +1,205 @@
+  offset: [number, number];
+  onMentionClick?: (username: string) => unknown;
+}
+
+interface UseUserMentionProps {
+  postId: string;
+  onInput: (content: string) => unknown;
+  commentRef?: MutableRefObject<HTMLDivElement>;
+}
+
+const isAlphaNumeric = (event: KeyboardEvent) => {
+  const match = event.key.match(/^[a-z0-9]+$/i);
+
+  return match && match[0].length === 1;
+};
+
+const IGNORE_KEY = ['Shift', 'CapsLock'];
+const shouldIgnoreKey = (event: KeyboardEvent) =>
+  IGNORE_KEY.indexOf(event.key) !== -1;
+
+const isBreakLine = (node: ChildNode) => !node.nodeValue;
+
+const getNode = (
+  el: HTMLElement,
+  query: string,
+): [ChildNode, { x: number; y: number }] => {
+  const index = Array.from(el.childNodes).findIndex((child) => {
+    const element = child.nodeValue ? child : child.childNodes[0];
+
+    if (isBreakLine(element)) {
+      return false;
+    }
+
+    const strings = element.nodeValue.split(' ');
+
+    return strings.some((string) => string === query);
+  });
+  const child = el.childNodes[index];
+  const node = child.nodeValue ? child : child.childNodes[0];
+  const start = node.nodeValue.indexOf(query);
+
+  return [node, { x: start, y: index }];
+};
+
+function setCaret(el: HTMLElement, replacement: string) {
+  const range = document.createRange();
+  const sel = window.getSelection();
+  const [node, { x }] = getNode(el, replacement);
+
+  range.setStart(node, x + replacement.length);
+  range.collapse(true);
+
+  sel.removeAllRanges();
+  sel.addRange(range);
+}
+
+function getCaretPost(el: Element): [number, number] {
+  const sel = window.getSelection();
+  const row = Array.from(el.childNodes).findIndex((child) => {
+    const element = child.nodeValue ? child : child.childNodes[0];
+
+    return sel.anchorNode === element;
+  });
+  return [sel.anchorOffset, row];
+}
+
+const getPossibleMentions = (el: HTMLElement) => {
+  const content = el.innerText.split(' ');
+  const mentions = content.filter((word) => word[0] === '@' && word.length > 1);
+
+  return mentions;
+};
+
+export function useUserMention({
+  postId,
+  commentRef,
+  onInput,
+}: UseUserMentionProps): UseUserMention {
+  const key = ['user-mention', postId];
+  const { user } = useContext(AuthContext);
+  const client = useQueryClient();
+  const [selected, setSelected] = useState(0);
+  const [offset, setOffset] = useState<[number, number]>([0, 0]);
+  const [mentions, setMentions] = useState<string[]>([]);
+  const [mentionQuery, setMentionQuery] = useState<string>();
+  const { data = { recommendedMentions: [] }, refetch } =
+    useQuery<RecommendedMentionsData>(
+      key,
+      () =>
+        request(`${apiUrl}/graphql`, RECOMMEND_MENTIONS_QUERY, {
+          postId,
+          query: mentionQuery,
+        }),
+      {
+        enabled: !!user,
+        refetchOnWindowFocus: false,
+        refetchOnMount: false,
+      },
+    );
+  const { recommendedMentions: mentionUsers } = data;
+  const fetchUsers = useDebounce(refetch, 300);
```

Comment:
Is this the same way we handle throttling on the search page? I think we should use the same mechanism

Reply:
It is currently using a timeout. Though their process flow is the same, should we update the search field implementations to utilize this hook?

Reply:
On a different ticket perhaps

Reply:
Alright, let me create one for it.